lambdas/refresh.js

```diff
@@ -0,0 +1,155 @@
+              'Authorization': `OAuth ${AccessToken}`
+            }
+          }, res => {
+            res.setEncoding('utf8')
+            const body = []
+            res.on('data', data => {
+              body.push(data)
+            })
+            res.on('end', () => {
+              const { Username } = JSON.parse(body.join('')).AccessingUser
+              const identityClient = new CognitoIdentity()
+              identityClient
+                .getOpenIdTokenForDeveloperIdentity({
+                  IdentityPoolId: cognitoIdentityPoolId,
+                  Logins: {
+                    [cognitoDeveloperProvider]: `${Username}@devzone`
+                  },
+                  TokenDuration: 86400 // Token lifetime 24hrs
+                })
+                .promise()
+                .then(({ IdentityId, Token }) => {
+                  const cognitosync = new CognitoSync()
+                  cognitosync
+                    .listRecords({
+                      DatasetName: 'identityInfo',
+                      IdentityId,
+                      IdentityPoolId: cognitoIdentityPoolId
+                    })
+                    .promise()
+                    .then((res) => {
+                      const { SyncSessionToken, Records } = res
+                      return cognitosync
+                        .updateRecords({
+                          DatasetName: 'identityInfo',
+                          IdentityId,
+                          IdentityPoolId: cognitoIdentityPoolId,
+                          SyncSessionToken,
+                          ClientContext: 'oauth-refresh',
+                          RecordPatches: [{
+                            Key: 'refresh_token',
+                            Op: 'replace',
+                            SyncCount: (Records.find(({ Key }) => Key === 'refresh_token') || {}).SyncCount || 0,
+                            Value: RefreshToken
+                          }]
+                        })
+                        .promise()
+                    })
+                  resolve({
+                    statusCode: 201,
+                    headers: {
+                      'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify({
+                      token: Token,
+                      refresh: RefreshToken
+                    })
+                  })
+                })
+                .catch(err => resolve({
+                  statusCode: 500,
+                  headers: {
+                    'Content-Type': 'application/json'
+                  },
+                  body: JSON.stringify(err)
+                }))
+            })
+          })
+        })
+      })
+    req.on('error', err => resolve({
+      statusCode: 500,
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(err)
+    }))
+    req.write(postData)
+    req.end()
+  })
+
+  callback(null, refreshResult)
```

Comment:
Return the result here:
```suggestion
  return refreshResult
```